File: projects/android/Sample/app/src/main/java/com/ezored/sample/ui/fragment/base/BaseListFragment.java
package com.ezored.sample.ui.fragment.base;

import android.support.v4.content.ContextCompat;
import android.support.v7.widget.DefaultItemAnimator;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.ezored.sample.R;
import com.yqritc.recyclerviewflexibledivider.HorizontalDividerItemDecoration;

import java.util.ArrayList;

public class BaseListFragment<T> extends BaseFragment {

    protected ArrayList<T> listData;
    protected RecyclerView rvList;
    protected LinearLayoutManager layoutManager;

    @Override
    protected void createAll(View view) {
        super.createAll(view);

        layoutManager = getLayoutManager();

        rvList = view.findViewById(R.id.rv_list);
        rvList.setLayoutManager(layoutManager);
        rvList.setItemAnimator(new DefaultItemAnimator());
        rvList.setNestedScrollingEnabled(false);
        rvList.setHasFixedSize(true);
        rvList.addItemDecoration(new HorizontalDividerItemDecoration.Builder(getContext()).color(ContextCompat.getColor(getContext(), R.color.grey_200)).build());
    }

    protected LinearLayoutManager getLayoutManager() {
        LinearLayoutManager layoutManager = new LinearLayoutManager(getContext());
        layoutManager.setOrientation(LinearLayoutManager.VERTICAL);
        return layoutManager;
    }

    protected RecyclerView.Adapter getAdapter() {
        // need implement on child fragment
        return null;
    }

    protected void updateAdapter() {
        rvList.setAdapter(getAdapter());
    }

    @Override
    protected int getFragmentLayout() {
        return R.layout.fragment_base_list;
    }

    @Override
    protected String getScreenNameForAnalytics() {
        return "Base list";
    }

}

File: projects/android/Sample/app/src/main/java/com/ezored/sample/ui/fragment/base/BaseListFragment.kt
package com.ezored.sample.ui.fragment.base

import android.view.View
import androidx.core.content.ContextCompat
import androidx.lifecycle.MutableLiveData
import androidx.recyclerview.widget.DefaultItemAnimator
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.Adapter
import com.ezored.sample.R

open class BaseListFragment<T> : BaseFragment() {

    protected var listData: MutableLiveData<ArrayList<T>>? = null
    protected lateinit var rvList: RecyclerView
    protected lateinit var layoutManager: LinearLayoutManager
    protected open lateinit var adapter: Adapter<*>

    override val fragmentLayout: Int
        get() = R.layout.fragment_base_list

    override val screenNameForAnalytics: String?
        get() = "Base list"

    override fun createAll(view: View) {
        super.createAll(view)

        layoutManager = getLayoutManagerForList()

        rvList = view.findViewById(R.id.rv_list)

        rvList.let { rvList ->
            rvList.layoutManager = layoutManager
            rvList.itemAnimator = DefaultItemAnimator()
            rvList.isNestedScrollingEnabled = false
            rvList.setHasFixedSize(true)

            context?.let { context ->
                val divider = DividerItemDecoration(context, DividerItemDecoration.VERTICAL)

                divider.setDrawable(
                    ContextCompat.getDrawable(
                        context,
                        R.drawable.list_item_separator
                    )!!
                )

                rvList.addItemDecoration(divider)
            }
        }
    }

    protected fun getLayoutManagerForList(): LinearLayoutManager {
        val layoutManager = LinearLayoutManager(context)
        layoutManager.orientation = RecyclerView.VERTICAL
        return layoutManager
    }

    protected fun updateAdapter() {
        rvList.adapter = adapter
    }
}
